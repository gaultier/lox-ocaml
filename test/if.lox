if (true) print "then"; else print "should not happen"; // expect: then

if (1 < 2) print "1 is smaller than 2"; // expect: 1 is smaller than 2

if (1 <= 1) print "1 is smaller or equal than 1"; else print "should not happen"; // expect: 1 is smaller or equal than 1

if (nil) print "should not happen"; else print "nil is falsey"; // expect: nil is falsey

if (2 > 1) {
   print "hello"; // expect: hello
   print "world"; // expect: world
}

if (nil) {
  print "should not happen";
} else {
  print "joe"; // expect: joe
  print "jane"; // expect: jane
}
