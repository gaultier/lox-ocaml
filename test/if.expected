then
1 is smaller than 2
1 is smaller of equal than 1
nil is falsey
hello
world
joe
jane
